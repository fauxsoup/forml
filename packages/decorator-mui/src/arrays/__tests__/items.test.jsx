import ArrayComponent from "../items";
import Context from "@forml/context";
import React from "react";
import { render } from "@testing-library/react";
import * as decorator from "../";

describe("renders", function() {
    let form;
    let title = "title";
    let description = "description";
    let localizer;

    beforeEach(function() {
        form = { type: "array", items: [{ key: [] }] };
        localizer = { getLocalizedString: jest.fn(id => id) };
    });

    describe("with form options", function() {
        let fields = {
            disableGutters: [true, false],
            disablePadding: [true, false],
            icon: ['person', 'favorite']
        };

        Object.keys(fields).forEach(function(field) {
            fields[field].forEach(function(value) {
                describe(`${field}`, function() {
                    test(`${value}`, function() {
                        form = { ...form, [field]: value };
                        const { container } = render(
                            <Context.Provider value={{ decorator, localizer }}>
                                <ArrayComponent
                                    form={form}
                                    title={title}
                                    description={description}
                                />
                            </Context.Provider>
                        );

                        expect(container).toMatchSnapshot();
                    });
                });
            });
        });
    });

    test("with title and description", function() {
        const { container } = render(
            <Context.Provider value={{ decorator, localizer }}>
                <ArrayComponent form={form} title={title} description={description} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test("with title and no description", function() {
        const { container } = render(
            <Context.Provider value={{ decorator, localizer }}>
                <ArrayComponent form={form} title={title} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test("with description and no title", function() {
        const { container } = render(
            <Context.Provider value={{ decorator, localizer }}>
                <ArrayComponent form={form} description={title} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test("with no title or description", function() {
        const { container } = render(
            <Context.Provider value={{ decorator, localizer }}>
                <ArrayComponent form={form} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });
});
