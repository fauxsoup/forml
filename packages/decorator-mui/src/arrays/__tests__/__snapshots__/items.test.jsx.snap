// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with description and no title 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-41 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-list-44 MuiList-dense"
    >
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
          edge="start"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            view_list
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            title
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root makeStyles-controls-45 MuiListItem-dense MuiListItem-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          edge="end"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                add
              </span>
            </span>
            Add
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders with form options disableGutters false 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-6 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-list-9 MuiList-dense"
    >
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
          edge="start"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            view_list
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            description
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root makeStyles-controls-10 MuiListItem-dense MuiListItem-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          edge="end"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                add
              </span>
            </span>
            Add
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders with form options disableGutters true 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-1 makeStyles-disableGutters-3 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-list-4 MuiList-dense"
    >
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
          edge="start"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            view_list
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            description
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root makeStyles-controls-5 MuiListItem-dense MuiListItem-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          edge="end"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                add
              </span>
            </span>
            Add
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders with form options disablePadding false 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-16 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-list-19 MuiList-dense"
    >
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
          edge="start"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            view_list
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            description
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root makeStyles-controls-20 MuiListItem-dense MuiListItem-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          edge="end"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                add
              </span>
            </span>
            Add
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders with form options disablePadding true 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-11 makeStyles-disablePadding-12 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-list-14 MuiList-dense"
    >
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
          edge="start"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            view_list
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            description
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root makeStyles-controls-15 MuiListItem-dense MuiListItem-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          edge="end"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                add
              </span>
            </span>
            Add
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders with form options icon favorite 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-26 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-list-29 MuiList-dense"
    >
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
          edge="start"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            favorite
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            description
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root makeStyles-controls-30 MuiListItem-dense MuiListItem-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          edge="end"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                add
              </span>
            </span>
            Add
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders with form options icon person 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-21 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-list-24 MuiList-dense"
    >
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
          edge="start"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            person
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            description
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root makeStyles-controls-25 MuiListItem-dense MuiListItem-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          edge="end"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                add
              </span>
            </span>
            Add
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders with no title or description 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-46 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-list-49 MuiList-dense"
    >
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
          edge="start"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            view_list
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        />
      </li>
      <li
        class="MuiListItem-root makeStyles-controls-50 MuiListItem-dense MuiListItem-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          edge="end"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                add
              </span>
            </span>
            Add
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders with title and description 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-31 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-list-34 MuiList-dense"
    >
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
          edge="start"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            view_list
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            description
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root makeStyles-controls-35 MuiListItem-dense MuiListItem-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          edge="end"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                add
              </span>
            </span>
            Add
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders with title and no description 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-36 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-list-39 MuiList-dense"
    >
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
          edge="start"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            view_list
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        />
      </li>
      <li
        class="MuiListItem-root makeStyles-controls-40 MuiListItem-dense MuiListItem-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          edge="end"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                add
              </span>
            </span>
            Add
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;
